lib/Sub/Override.pm

```diff
@@ -82,6 +82,7 @@ sub replace {
     {
         no strict 'refs';
         $self->{$sub_to_replace} ||= *$sub_to_replace{CODE};
+        set_prototype(\&$new_sub, prototype($self->{$sub_to_replace}));
```

Comment:
The problem with this is that it could break a lot of existing code. As I understand it, you can currently replace a prototyped subroutine with a non-prototyped one and everything will just work no matter what we pass the new routine. If we suddenly come along and insist on the replacements following the original prototypes... eeep!

The 'wrap' update was more desireable because a) it was a new feature and few people would have started using it, and b) if the routine being replaced was prototyped, it would cause problems with the wrap, so needed to be addressed.

Perhaps we could implement a class-static setting that says 'forward_prototypes' or something, that is off by default. We could use this for all the override routines, not just 'replace'. Thoughts?

Let me know if I have something misunderstood.

Reply:
Ah, it looks like I assumed incorrectly... the prototypes are actually used in the new sub. I should have checked first, just didn't occur to me that it didn't work that way.

Reply:
I've merged #19 which had a pile of doc updates. If you're willing, how about catching up this PR and add your prototyping to inject/inherit as well?

Reply:
...although inject wouldn't need it, since it's new. :-0